receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/factory.go

```diff
@@ -54,6 +55,25 @@ func (f *Factory) CreateMetricsScraper(
 	config internal.Config,
 ) (scraperhelper.Scraper, error) {
 	cfg := config.(*Config)
+
+	if cfg.RootPath == "" {
+		inContainer := os.Getpid() == 1
+		for _, p := range []string{
+			"/.dockerenv",        // Mounted by dockerd when starting a container by default
+			"/run/.containerenv", // Mounted by podman as described here: https://github.com/containers/podman/blob/ecbb52cb478309cfd59cc061f082702b69f0f4b7/docs/source/markdown/podman-run.1.md.in#L31
+		} {
+			if _, err := os.Stat(p); err == nil {
+
+				break
+			}
+		}
```

Comment:
@MovieStoreGuy, I missed checking the PR after the latest update. It looks like this block doesn't do anything. Can you please remove it?

Reply:
Let me chase it up.